base/sparse/higherorderfns.jl

```diff
@@ -1061,8 +1062,8 @@ promote_containertype(::Type{Array}, ::Type{AbstractSparseArray}) = PromoteToSpa
 # mostly just disambiguates Array from the main containertype promotion mechanism
 # AbstractArray serves as a marker to shunt to the generic AbstractArray broadcast code
 _spcontainertype(x) = _containertype(x)
-_spcontainertype(::Type{<:Vector}) = Vector
-_spcontainertype(::Type{<:Matrix}) = Matrix
+_spcontainertype(::Type{<:AbstractVector}) = Vector
```

Comment:
Loosening these signatures would cause sparse `broadcast[!]` to capture a much broader array of type combinations than it does now. A conservative approach seems wiser. Instead just add a method for row vectors?